.golangci.yml

```diff
@@ -0,0 +1,36 @@
+linters-settings:
+  govet:
+    check-shadowing: true
+
+linters:
+  disable-all: true
+  enable:
+    - unused
+    - gosimple
+    - structcheck
+    - varcheck
+    - ineffassign
+    - deadcode
+    - unconvert
+    # TODO: enable these later
+    # - govet
+    # - gofmt
+    # - errcheck
+    # - staticcheck
+    # - unparam
+
+issues:
+  exclude-rules:
+    - linters:
+      # ignore warnings to simplify v, _ := to v :=
+      - gosimple
+      text: "S1005:"
+
+    - linters:
+      # ignore unused warnings from enterprise code
+      # add more as required.
+      - unused
+      text: "RedisSupplier|LocalCacheSupplier|Enterprise"
+
+  # TODO: slowly remove all warnings from the codebase and remove this.
+  new-from-rev: HEAD~
```

Comment:
The problem with this is, that a PR might contain multiple commits. In this case `golangci-lint` would only report the last one. I'm inclined to remove this check to remove this check, rely on https://golangci.com/r/github.com/mattermost/mattermost-server to find the new errors and gradually fix the current problems.

What do you think about this?

Reply:
Yep, I knew about this, but it seemed to be the best possible solution for now. The issue with golangci will be that it won't be enforced, and we will again fall back to slowly fixing the problems while there will be no check on new issues creeping in.

Reply:
>  The issue with using golangci service will be that it won't be enforced

It will! It adds a GH check, which we can require for merges to `master`. Or am I forgetting something?

Reply:
My bad ! I was thinking it would be like a passive check. But I see that you have already sent PRs to fix existing issues. Great stuff ! :+1: